libnavigation-base/src/main/java/com/mapbox/navigation/base/logger/Logger.kt

```diff
@@ -1,53 +1,56 @@
 package com.mapbox.navigation.base.logger
 
+import com.mapbox.navigation.base.logger.model.Message
+import com.mapbox.navigation.base.logger.model.Tag
+
 interface Logger {
     /**
      * Send a verbose log message and log the exception.
      *
-     * @param tag Used to identify the source of a log message.  It usually identifies
+     * @param msg is [Message] you would like logged.
+     * @param tag is [Tag] used to identify the source of a log message.  It usually identifies
      * the class or activity where the log call occurs.
-     * @param msg The message you would like logged.
      * @param tr An exception to log
      */
-    fun v(tag: String, msg: String, tr: Throwable? = null)
+    fun v(msg: Message, tag: Tag? = null, tr: Throwable? = null)
```

Comment:
Could you swap places of `msg` and `tag` as is done in the default implementation?

Reply:
Problem is that default parameters need to be in order. It's a Kotlin limitation 👀 https://kotlinlang.org/docs/reference/functions.html#default-arguments

> If a default parameter precedes a parameter with no default value, the default value can only be used by calling the function with [named arguments](https://kotlinlang.org/docs/reference/functions.html#named-arguments)

Reply:
I'd rather expose the more intuitive order of arguments than change the order just because of language limitations, especially since you can use named arguments to get the desired result anyway.